boot/bootutil/src/ed25519_psa.c

```diff
@@ -165,4 +165,32 @@ int exec_revoke(void)
     return ret;
 }
 #endif /* CONFIG_BOOT_KMU_KEYS_REVOCATION */
+
+void nrf_crypto_keys_housekeeping(void)
+{
+    psa_status_t status;
+
+    /* We will continue through all keys, even if we have error while
+     * processing any of it. Only doing BOOT_LOG_DBG, as we do not
+     * really want to inform on failures to lock.
+     */
+    for (int i = 0; i < CONFIG_BOOT_SIGNATURE_KMU_SLOTS; ++i) {
+        psa_key_attributes_t attr;
+
+        status = psa_get_key_attributes(kmu_key_ids[i], &attr);
+        BOOT_LOG_DBG("KMU key 0x%x(%d) attr query status == %d",
+                     kmu_key_ids[i], i, status);
+
+        if (status == PSA_SUCCESS) {
+            status = cracen_kmu_block(&attr);
+            BOOT_LOG_DBG("KMU key lock status == %d", status);
+        }
+
+        status = psa_purge_key(kmu_key_ids[i]);
```

Comment:
Commenting here as we have merged an API that is irrelevant for KMU support 
https://arm-software.github.io/psa-api/crypto/1.1/api/keys/management.html#c.psa_purge_key

This type of API is only required when [PSA_KEY_USAGE_CACHE](https://arm-software.github.io/psa-api/crypto/1.1/api/keys/policy.html#c.PSA_KEY_USAGE_CACHE) is in use, which is not the case for NSIB or MCUboot

There is no key caching functionality forwarded to PSA crypto driver wrappers (psa_crypto_driver_wrapper_purge-key doesn't exists). Likewise there is nothing in the thin PSA core used by [bl2](https://github.com/nrfconnect/sdk-trusted-firmware-m/blob/main/bl2/src/thin_psa_crypto_core.c)